catchup/service.go

```diff
@@ -710,6 +728,22 @@ func (s *Service) fetchRound(cert agreement.Certificate, verifier *agreement.Asy
 				return
 			default:
 			}
+			if err == errNoBlockForRound {
+				// If a peer does not have the block after few attempts it probably has not persisted the block yet.
+				// Give it some time to persist the block and try again.
+				// Quit if the the same peer errs more than errNoBlockForRoundThreshold times.
+				if count, ok := peerErrors[peer]; ok {
+					if count == errNoBlockForRoundThreshold {
```

Comment:
Do we want to do this only once before returning from the function (and falling back to the 17s timeout), or perhaps do another few attempts and try again?

Reply:
I think a single retry is OK, it is basically to give the remote a bit time to see the cert vote (if we saw it before) and call `EnsureBlock` in agreement, or to give some time to call `EnsureBlock` if we saw it simultaneously.
Well, I guess technically we need to wait 2 * small lambda time but it looks too much for a single sleep, so waiting few times but no more than 2 * SL looks reasonable. Updating the PR.